### ISSUE TITLE
When installing a new extension the Theme resets to default

### ISSUE THREAD

**Issue by MauriceElliott** [zed-industries/zed#14555]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Fresh install of Zed by building from source on windows
Opened a repository that contained Terraform files
Opened a terraform file
The pop up for installing terraform language support displayed
Clicked okay to install
The theme then reset to default.

### Environment

Zed: v0.145.0 (Zed Dev ef20afa9a43fccbbf4fca1bee04d9725373363fb)
OS: Windows 10.0.19045
Memory: 15.7 GiB
Architecture: x86_64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

2024-07-16T11:38:41.871071+01:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-07-16T10:38:41.661Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-07-16T10:38:41.661Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-07-16T11:38:44.3276931+01:00 [INFO] Installing default prettier and plugins: [("prettier", "3.3.3")]
2024-07-16T11:38:44.3284481+01:00 [INFO] Node runtime install_if_needed
2024-07-16T11:38:44.6114617+01:00 [INFO] Creating buffer 0x27d46ea1698 of size 4096, name 'chunk-3', handle 5
2024-07-16T11:38:44.6654799+01:00 [INFO] Using surface present mode IMMEDIATE
2024-07-16T11:38:44.7297348+01:00 [INFO] Creating buffer 0x27d46ea2b08 of size 5488, name 'chunk-4', handle 6
2024-07-16T11:38:44.730532+01:00 [INFO] Creating buffer 0x27d46e9f488 of size 4096, name 'chunk-5', handle 7
2024-07-16T11:38:47.2767245+01:00 [INFO] Creating buffer 0x27d46e9fb58 of size 25872, name 'chunk-6', handle 8
2024-07-16T11:38:47.2858333+01:00 [INFO] Creating buffer 0x27d46ea08f8 of size 5376, name 'chunk-7', handle 9
2024-07-16T11:38:47.2864335+01:00 [INFO] Creating buffer 0x27d46ea0fc8 of size 25872, name 'chunk-8', handle 10
2024-07-16T11:38:49.0781647+01:00 [INFO] Initialized prettier with plugins: {}
2024-07-16T11:39:03.5901024+01:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-07-16T11:39:03.6008257+01:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-07-16T11:39:03.6091484+01:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-07-16T11:39:12.4343297+01:00 [INFO] Opening main db
2024-07-16T11:39:12.5392704+01:00 [INFO] Creating buffer 0x27d46ea2438 of size 83888, name 'chunk-9', handle 11
2024-07-16T11:39:12.5464009+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:39:12.547177+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:39:12.5482888+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:39:12.548751+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:39:12.5769931+01:00 [INFO] Creating buffer 0x27d46f3d398 of size 7056, name 'chunk-10', handle 12
2024-07-16T11:39:12.5777485+01:00 [INFO] Creating buffer 0x27d46f3ccc8 of size 24192, name 'chunk-11', handle 13
2024-07-16T11:39:12.5783518+01:00 [INFO] Creating buffer 0x27d46f3b858 of size 83888, name 'chunk-12', handle 14
2024-07-16T11:42:35.3117452+01:00 [INFO] Creating texture 0x27cbf1e9258 of size 1024x1024x1 and format R16Float, name 'atlas', handle 15
2024-07-16T11:42:35.3123178+01:00 [INFO] Creating buffer 0x27d46f3da68 of size 84000, name 'chunk-13', handle 16
2024-07-16T11:42:35.996123+01:00 [INFO] Creating buffer 0x27d46f3aab8 of size 84112, name 'chunk-14', handle 17
2024-07-16T11:42:36.2286334+01:00 [INFO] Creating buffer 0x27d46f3e138 of size 92512, name 'chunk-15', handle 18
2024-07-16T11:42:36.2433148+01:00 [INFO] Creating buffer 0x27d46f3fc78 of size 98896, name 'chunk-16', handle 19
2024-07-16T11:42:36.2624721+01:00 [INFO] Creating buffer 0x27d46f3eed8 of size 120288, name 'chunk-17', handle 20
2024-07-16T11:42:36.2938722+01:00 [INFO] Creating buffer 0x27d46f3e808 of size 126224, name 'chunk-18', handle 21
2024-07-16T11:42:36.5122182+01:00 [INFO] Creating buffer 0x27d46f3f5a8 of size 147952, name 'chunk-19', handle 22
2024-07-16T11:42:36.5447962+01:00 [INFO] Creating buffer 0x27d46f3b188 of size 153888, name 'chunk-20', handle 23
2024-07-16T11:42:37.0783409+01:00 [INFO] Creating buffer 0x27d46f40348 of size 160160, name 'chunk-21', handle 24
2024-07-16T11:45:39.2122663+01:00 [INFO] set status on client 0: Authenticating
2024-07-16T11:45:39.2152818+01:00 [ERROR] Element not found. (0x80070490)
2024-07-16T11:45:45.229697+01:00 [INFO] set status on client 194752: Connecting
2024-07-16T11:45:45.48152+01:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-07-16T11:45:45.840275+01:00 [INFO] add connection to peer
2024-07-16T11:45:45.8406945+01:00 [INFO] add_connection;
2024-07-16T11:45:45.8414256+01:00 [INFO] waiting for server hello
2024-07-16T11:45:45.8447306+01:00 [INFO] got server hello
2024-07-16T11:45:45.845063+01:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 452, id: 783070 })
2024-07-16T11:45:45.8456973+01:00 [INFO] set status on client 194752: Connected { peer_id: PeerId { owner_id: 452, id: 783070 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-07-16T11:45:46.5765658+01:00 [INFO] Creating texture 0x27d46fc7038 of size 1024x1024x1 and format Bgra8UnormSrgb, name 'atlas', handle 25
2024-07-16T11:45:52.3232708+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:45:52.3235269+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:45:52.3512973+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-16T11:45:52.355389+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-16T11:45:52.3693142+01:00 [INFO] Creating a descriptor pool for at most 256 sets
2024-07-16T11:45:52.3858381+01:00 [INFO] Creating a descriptor pool for at most 256 sets
2024-07-16T11:45:59.8089139+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:45:59.8093268+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:02.6408815+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:02.6411575+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:07.0716585+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:07.0720486+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:10.4771194+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:10.4774791+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:15.0445128+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:15.0449206+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:15.6401999+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:15.640738+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:16.1988884+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:16.1991667+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:16.4446715+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:16.4449656+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:16.640709+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:16.6409919+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:16.9111278+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:16.9113784+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:17.1605847+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:17.1609374+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:17.4801505+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:17.4804502+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:17.7837757+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:17.784034+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:18.1207617+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:18.121041+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:18.428491+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:18.4287903+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:18.7347285+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:18.7349904+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:19.0446662+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:19.0449313+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:19.359513+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:19.3598568+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:19.6515662+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:19.6518777+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:19.9739305+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:19.9742084+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:20.2988001+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:20.2991503+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:20.59964+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:20.5999509+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:20.9067234+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:20.9070477+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:21.4602193+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:21.4605459+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:22.2788901+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:22.2792097+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:46:22.6228375+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T11:46:22.6230931+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T11:53:42.5287573+01:00 [ERROR] Error { code: HRESULT(0x00000000), message: "The operation completed successfully." }
2024-07-16T12:00:11.6006328+01:00 [INFO] Creating buffer 0x27d46f40a18 of size 160384, name 'chunk-22', handle 26
2024-07-16T12:00:13.9062884+01:00 [INFO] Creating buffer 0x27d46f410e8 of size 162960, name 'chunk-23', handle 27
2024-07-16T12:00:42.4031229+01:00 [INFO] installing extension terraform latest version
2024-07-16T12:00:43.4966872+01:00 [INFO] rebuilt extension index in 19.3705ms
2024-07-16T12:00:43.4970439+01:00 [INFO] extensions updated. loading 1, reloading 0, unloading 0
2024-07-16T12:00:43.4998259+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:00:43.5001975+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T12:00:43.6968277+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:00:43.6971858+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.6993054+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:00:43.6997187+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.7023338+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:00:43.7026061+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.7048745+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:00:43.7051987+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.7072784+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:00:43.7076767+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.7083354+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:00:43.7087324+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T12:00:43.7090502+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.7093781+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.7096064+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.7098366+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.7100986+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:00:43.7235244+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:00:43.7239706+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T12:00:43.7243231+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:00:43.7246427+01:00 [INFO] starting language server "terraform-ls", path: "C:\\repos\\github_repos\\az-github-runners", id: 1
2024-07-16T12:00:43.7254754+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:00:43.7257401+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:00:43.726079+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:00:43.7263208+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:00:45.2063534+01:00 [INFO] starting language server. binary path: "C:\\Users\\Maurice.Elliott\\AppData\\Local\\Zed\\extensions\\work\\terraform\\terraform-ls-v0.34.0\\terraform-ls", working directory: "C:\\repos\\github_repos\\az-github-runners", args: ["serve"]
2024-07-16T12:01:01.1143143+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:01.1145959+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T12:01:01.1147762+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:01.1149614+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:01.1151886+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:01.115374+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:01.1155794+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:02.8570327+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:02.8573625+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T12:01:02.8576453+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:02.8579423+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:02.8582423+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:02.8585318+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:02.8588378+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:11.1603893+01:00 [INFO] installing extension terraform latest version
2024-07-16T12:01:11.935501+01:00 [INFO] rebuilt extension index in 18.8919ms
2024-07-16T12:01:11.9358658+01:00 [INFO] extensions updated. loading 0, reloading 1, unloading 0
2024-07-16T12:01:11.9388131+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:11.9391013+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T12:01:12.1488614+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:12.1493773+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:01:12.1517103+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:12.1521376+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:01:12.1545604+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:12.1550421+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:01:12.1573187+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:12.1576874+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:01:12.1604252+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:12.1607977+01:00 [INFO] starting language servers for Terraform: 
2024-07-16T12:01:12.1794088+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:12.1796991+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T12:01:12.179979+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:12.1803244+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:12.1805469+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:12.1808055+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:12.1810497+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:17.8961299+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:17.896375+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T12:01:17.8965287+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:17.8967125+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:17.896956+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:17.8971185+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:17.8973412+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:19.3806917+01:00 [INFO] Initializing default prettier with plugins {}
2024-07-16T12:01:19.3809295+01:00 [INFO] starting language servers for Shell Script: 
2024-07-16T12:01:19.3810813+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:19.3812534+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:19.3814648+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:19.3816272+01:00 [INFO] starting language servers for Terraform: terraform-ls
2024-07-16T12:01:19.3817897+01:00 [INFO] starting language servers for Terraform: terraform-ls
</pre></details>

**Comment by notpeter** [comment]
Can you verify that your settings.json is valid JSON?  If there is an error parsing settings.json (e.g. extra unclosed `{`, etc) Zed may revert to default settings which could revert your theme selection.

**Comment by MauriceElliott** [comment]
> Can you verify that your settings.json is valid JSON? If there is an error parsing settings.json (e.g. extra unclosed `{`, etc) Zed may revert to default settings which could revert your theme selection.

// Zed settings
//
// For information on how to configure Zed, see the Zed
// documentation: https://zed.dev/docs/configuring-zed
//
// To see all of Zed's default settings without changing your
// custom settings, run the `open default settings` command
// from the command palette or from `Zed` application menu.
{
  "ui_font_size": 16,
  "buffer_font_size": 16,
  "theme": {
    "mode": "system",
    "light": "One Light",
    "dark": "Andromeda"
  }
}

**Comment by MauriceElliott** [comment]
@notpeter 
Looks fine to me, its basically a fresh install. I noticed it also did the same yesterday after creating this issue when opening a yaml file, which implies to me it was the switching of the language server for the currently opened file. Once I switched it back to my chosen theme, it didn't happen again.

However, literally just now as I've opened the settings.json, its reverted to the default again. Must be tied to the syntax highlighting in some way?
